src/m365/teams/commands/team/team-list.spec.ts

```diff
@@ -217,31 +198,34 @@ describe(commands.TEAM_LIST, () => {
 
   it('lists Microsoft Teams in the tenant (debug)', (done) => {
     sinon.stub(request, 'get').callsFake((opts) => {
-      if (opts.url === `https://graph.microsoft.com/beta/groups?$filter=resourceProvisioningOptions/Any(x:x eq 'Team')&$select=id,displayName,description`) {
+      if (opts.url === `https://graph.microsoft.com/v1.0/groups?$select=id,displayName,description,resourceProvisioningOptions`) {
         return Promise.resolve({
           "value": [
             {
               "id": "02bd9fd6-8f93-4758-87c3-1fb73740a315",
               "description": "Team 1 description",
-              "displayName": "Team 1"
+              "displayName": "Team 1",
+              "resourceProvisioningOptions": ["Team"]
             },
             {
               "id": "13be6971-79db-4f33-9d41-b25589ca25af",
               "description": "Team 2 description",
-              "displayName": "Team 2"
+              "displayName": "Team 2",
+              "resourceProvisioningOptions": ["Team"]
             },
             {
               "id": "8090c93e-ba7c-433e-9f39-08c7ba07c0b3",
               "description": "Team 3 description",
-              "displayName": "Team 3"
+              "displayName": "Team 3",
+              "resourceProvisioningOptions": ["Stream"]
             }
           ]
         });
       }
-      else if ((opts.url as string).startsWith(`https://graph.microsoft.com/beta/teams/`)) {
+      else if ((opts.url as string).startsWith(`https://graph.microsoft.com/v1.0/teams/`)) {
         const id: string = (<string>opts.url).substring((<string>opts.url).lastIndexOf(`/`) + 1);
-        return Promise.resolve({
-          "@odata.context": "https://graph.microsoft.com/beta/$metadata#teams/$entity",
+        return Promise.resolve([{
```

Comment:
This test case is failing because, in this function in actual is resolving JSON object in the original method. In your case, yoj are returning JSON array and hence it is causing the subsequent method to fail. Rather than you returning JSON array, you can return JSON object and this will work without any issue. So your code may look something like below.

```
else if ((opts.url as string).startsWith(`https://graph.microsoft.com/v1.0/teams/`)) {
        const id: string = (<string>opts.url).substring((<string>opts.url).lastIndexOf(`/`) + 1);
        return Promise.resolve({
          "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#teams/$entity",
          "id": id,
          "webUrl": "https://teams.microsoft.com/l/team/19:a5c6eccad3fb401997756a1501d561aa%40thread.skype/conversations?groupId=8090c93e-ba7c-433e-9f39-08c7ba07c0b3&tenantId=dcd219dd-bc68-4b9b-bf0b-4a33a796be35",
          "isArchived": false,
          "memberSettings": {
            "allowCreateUpdateChannels": true,
            "allowDeleteChannels": true,
            "allowAddRemoveApps": true,
            "allowCreateUpdateRemoveTabs": true,
            "allowCreateUpdateRemoveConnectors": true
          },
          "guestSettings": {
            "allowCreateUpdateChannels": false,
            "allowDeleteChannels": false
          },
          "messagingSettings": {
            "allowUserEditMessages": false,
            "allowUserDeleteMessages": false,
            "allowOwnerDeleteMessages": false,
            "allowTeamMentions": true,
            "allowChannelMentions": true
          },
          "funSettings": {
            "allowGiphy": true,
            "giphyContentRating": "moderate",
            "allowStickersAndMemes": true,
            "allowCustomMemes": false
          }
        });
      }
```
Totally understand the confusion for you in this case since you are having multiple promises. Eventhough promises are multiple, all it returns is a single object and all we need to do is mock it as a `Promise.resolve` of single JSON object. 

Hope it is clear for you now